Makefile

```diff
@@ -8,16 +10,32 @@ ONOS_CONFIG_VERSION ?= latest
 build-tools:=$(shell if [ ! -d "./build/build-tools" ]; then cd build && git clone https://github.com/onosproject/build-tools.git; fi)
 include ./build/build-tools/make/onf-common.mk
 
+mod-update: # @HELP Download the dependencies to the vendor folder
+	go mod tidy
+	go mod vendor
+
+mod-lint: mod-update # @HELP ensure that the required dependencies are in place
+	# dependencies are vendored, but not committed, go.sum is the only thing we need to check
+	bash -c "diff -u <(echo -n) <(git diff go.sum)"
+
+local-protos: # @HELP Copies a local version of the onos-api dependency into the vendor directory
```

Comment:
What about other deps? I mean can we replace onos-lib-go, sdk, etc without any problems with this PR or it is just onos-api

Reply:
As of now there is support for `onos-api` only, let me update to support the others as well